hack/release.sh

```diff
@@ -76,3 +76,15 @@ git tag --annotate --message "v${RELEASE_VERSION} Release" "v${RELEASE_VERSION}"
 
 printf "\033[36m==> %s\033[0m\n" "Push tag release v${RELEASE_VERSION}"
 git push origin v${RELEASE_VERSION}
+
+# Bump the next version in version.go
+NEXT_VERSION=$(echo "${RELEASE_VERSION}" | sed 's/^v//' | awk -F'[ .]' '{print $1"."$2+1".0"}')
+sed -i -E 's|v'${RELEASE_VERSION}'|v'${NEXT_VERSION}'-alpha|g' version/version.go
+
+# Commit changes
+printf "\033[36m==> %s\033[0m\n" "Bump version to v${NEXT_VERSION}-alpha"
+git add version/version.go
+git commit -m "Bump version to v${NEXT_VERSION}-alpha"
```

Comment:
It's a bit odd to see versions being hardcoded inside the code. I don't see this being done on any other services. IMO, code should be concerned with just the task it's doing and not contain a hardcoded version string in it. That can be done via git and then injected at runtime (exactly like we are doing via ldflags).

So my suggestion would be to drop the modification of the version file and just dynamically inject the version during building.

Reply:
I always put a hardcoded dev version in my projects, mainly to give visibility of the _version_ if someone `go get` rather than download the binary (released or rc or ...)
I can see the point here, because most probably no one will ever _go get_ this. But on the other hand that variable never gets changed manually and the release script changes it and also the injection upon release will override that anyways. So I'm kinda 2-ish/5 to keep it.

Reply:
The visibility of the version is not the responsibility of the code IMO. I'd be interested to see if this is documented as a best practice somewhere.

If it's a library, the version is tracked in the go.mod file, otherwise if it's a main binary, it is tracked in git. Exactly how it's done in all of our projects.

I think just embedding the version from git should be good enough.